core/prototypes/list.py

```diff
@@ -313,29 +313,35 @@ def index(self, *args, **kwargs):
 		"""
 			Default, SEO-Friendly fallback for view and list.
 
-			:param args:
-			:param kwargs:
-			:return:
+			:param args: The first argument - if provided - is interpreted as seoKey.
+			:param kwargs: Used for the fallback list.
+			:return: The rendered entity or list.
 		"""
 		if args and args[0]:
 			# We probably have a Database or SEO-Key here
-			seoKey = "viur.viurActiveSeoKeys ="
-			skel = self.viewSkel().all(_excludeFromAccessLog=True).filter(seoKey, str(args[0]).lower()).getSkel()
+			seoKey = str(args[0]).lower()
+			skel = self.viewSkel().all(_excludeFromAccessLog=True).filter("viur.viurActiveSeoKeys =", seoKey).getSkel()
 			if skel:
 				db.currentDbAccessLog.get(set()).add(skel["key"])
 				if not self.canView(skel):
 					raise errors.Forbidden()
+				lang = currentLanguage.get()
+				# Check whether this is the current seo-key, otherwise redirect to it
+				if (
+					skel["viurCurrentSeoKeys"]
+					and (currentSeoKey := skel["viurCurrentSeoKeys"].get(lang))
+					and seoKey != currentSeoKey
+				):
+					modulePath = self.modulePath
```

Comment:
You should use the function seoUrlToEntry from https://github.com/viur-framework/viur-core/blob/master/core/utils.py#L153

Reply:
I didn't know this method, good to know. Thank you for this info.